libazureinit/src/config.rs

```diff
@@ -345,6 +403,7 @@ impl Config {
 
         if base_path.exists() {
             tracing::info!(path=?base_path, "Loading base configuration file");
+            check_azure_init_header(&base_path)?;
```

Comment:
we only want this header in config coming from user-data / custom-data

local config files are azure-init configuration by definition.  it's the user data we want to make sure it meant for azure-init before attempting to load it